src/components/validation/resources/http.rs

```diff
@@ -225,26 +235,56 @@ fn spawn_output_http_server(
     codec: ResourceCodec,
     output_tx: mpsc::Sender<Vec<Event>>,
     task_coordinator: &TaskCoordinator<Configuring>,
+    runner_metrics: &Arc<Mutex<RunnerMetrics>>,
 ) {
     // This HTTP server will wait for events to be sent by a sink, and collect them and send them on
     // via an output sender. We accept/collect events until we're told to shutdown.
 
     // First, we'll build and spawn our HTTP server.
     let decoder = codec.into_decoder();
 
-    let (_, http_server_shutdown_tx) =
-        spawn_http_server(task_coordinator, &config, move |request| {
+    let (_, http_server_shutdown_tx) = spawn_http_server(
+        task_coordinator,
+        &config,
+        runner_metrics,
+        move |request, output_runner_metrics| {
+            if request.headers().contains_key("Content-Length") {
+                let bytes = request
+                    .headers()
+                    .get("Content-Length")
+                    .unwrap()
+                    .to_str()
+                    .unwrap();
+                debug!("HTTP server external output resource received {bytes} bytes.");
+            }
```

Comment:
Seems like it would just be better to take the length of the body directly after you collect it down on line 265?

Less code, and doesn't break on streaming (chunked) responses.